cmd/powermanage/main_test.go

```
@@ -0,0 +1,321 @@
+	directory := filepath.Join(t.TempDir(), "powermanage")
+	a := &app{
+		stdin: strings.NewReader(""), stdout: &bytes.Buffer{}, stderr: &bytes.Buffer{},
+		httpClient: control.Client(), now: time.Now,
+		configPath: filepath.Join(directory, "config.json"), sessionPath: filepath.Join(directory, "session.json"),
+	}
+	require.NoError(t, writePrivateJSON(a.configPath, configFile{ServerURL: control.URL}))
+	a.openBrowser = func(loginURL string) error {
+		parsed, err := url.Parse(loginURL)
+		require.NoError(t, err)
+		callback := parsed.Query().Get("redirect_uri") + "?code=authorization-code&state=" + loginState
+		response, err := http.Get(callback)
```

golangci-lint (2.12.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)